src/common/moisture_sensor.rs

```diff
@@ -0,0 +1,48 @@
+use crate::common::analog::AnalogReader;
+use crate::common::sensor::GenericReadingsResult;
+use crate::common::sensor::Sensor;
+use crate::common::sensor::SensorResult;
+use crate::common::sensor::SensorT;
+use crate::common::sensor::TypedReadingsResult;
+use crate::common::status::Status;
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+pub struct MoistureSensor {
+    analog: Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>,
+}
+
+impl MoistureSensor {
+    pub fn new(analog: Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>) -> Self {
+        MoistureSensor { analog: analog }
+    }
+}
+
+impl Sensor for MoistureSensor {
+    fn get_generic_readings(&self) -> anyhow::Result<GenericReadingsResult> {
+        Ok(self
```

Comment:
I don't particularly like that each sensor that implements `Sensor` needs to re-implement `get_generic_readings` for types that can be serialized. I tried to do some trait magic to make it happen automatically, but I couldn't quite get there:

```
impl<T, U: SensorT<T>> Sensor for U
where
    SensorResult<T>: Into<prost_types::Value>,
{
    fn get_readings(&self) -> anyhow::Result<GenericReadingsResult> {
        let readings = self.get_readings_t()?;
        let bt: ::std::collections::BTreeMap<::prost::alloc::string::String, ::prost_types::Value> =
            readings
                .into_iter()
                .map(|v| (v.0, SensorResult::<T> { value: v.1 }.into()))
                .collect();
        Ok(GenericReadingsResult { fields: bt })
    }
}
```

Fails with:

```
error[E0207]: the type parameter `T` is not constrained by the impl trait, self type, or predicates
  --> src/common/sensor.rs:77:6
   |
77 | impl<T, U: SensorT<T>> Sensor for U
   |      ^ unconstrained type parameter
```

I believe this is because there is nothing here to ensure that a type doesn't implement both, say `SensorT<u32>` *and* `SensorT<u64>`, so which `get_readings_t` did we mean to call?